src/k2/httpproxy/HTTPProxy.cpp

```diff
@@ -262,6 +266,7 @@ HTTPProxy::_handleQuery(shd::QueryRequest&& request) {
         if (auto iter = _txns.find(request.timestamp); iter == _txns.end()) {
             return MakeHTTPResponse<shd::QueryResponse>(Txn_S410_Gone, shd::QueryResponse{});
         } else {
+            updateExpiry(iter->second);
```

Comment:
why update expiry here? We already updated when we received query

Reply:
If there are large number of pages in query, the client may take long time to retrieve data. During time the txn should be kept alive.